docs/guides/cortex-with-cassandra.md

```diff
@@ -0,0 +1,174 @@
+auth_enabled: false
+
+server:
+  http_listen_port: 9009
+
+  # Configure the server to allow messages up to 100MB.
+  grpc_server_max_recv_msg_size: 104857600
+  grpc_server_max_send_msg_size: 104857600
+  grpc_server_max_concurrent_streams: 1000
+
+distributor:
+  shard_by_all_labels: true
+  pool:
+    health_check_ingesters: true
+
+ingester_client:
+  grpc_client_config:
+    # Configure the client to allow messages up to 100MB.
+    max_recv_msg_size: 104857600
+    max_send_msg_size: 104857600
+    use_gzip_compression: true
+
+ingester:
+  lifecycler:
+    # The address to advertise for this ingester. Will be autodiscovered by
+    # looking up address on eth0 or en0; can be specified if this fails.
+    address: 127.0.0.1
+
+    # We want to start immediately and flush on shutdown.
+    join_after: 0
+    claim_on_rollout: false
+    final_sleep: 0s
+    num_tokens: 512
+
+    # Use an in memory ring store, so we don't need to launch a Consul.
+    ring:
+      kvstore:
+        store: inmemory
+      replication_factor: 1
+
+# Use cassandra as storage -for both index store and chunks store.
+schema:
+  configs:
+  - from: 2019-07-29
+    store: cassandra
+    object_store: cassandra
+    schema: v10
+    index:
+      prefix: index_
+      period: 168h
+
+storage:
+  cassandra:
+    addresses: LOCALHOST # configure cassandra addresses here.
+    keyspace: KEYSPACE   # configure desired keyspace here.
+```
+
+Running cortex as an image
+
+```
+docker run -d --name=cortex -v $(pwd)/single-process-config.yaml:/etc/single-process-config.yaml -p 9009:9009  quay.io/cortexproject/cortex -config.file=/etc/single-process-config.yaml
+```
+
+Running cortex as a executable by building it from source
+
+clone the cortex project by using
+
+```
+git clone https://github.com/cortexproject/cortex.git
+```
+
+Build the cortex from source using
+
+```
+go build ./cmd/cortex
+```
+
+Now run the cortex using the executable generated and by passing single-process-config.yaml that we built in the previous steps.
+
+```
+./cortex -config.file=./single-process-config.yaml
+```
+
+Now the cortex is running on http://localhost:9009
+
+Add remote write configuration for your prometheus config file. This writes the metrics data to your cortex. 
+
+```
+remote_write:
+   - url: http://localhost:9009/api/prom/push
+```
+
+Run grafana to visualise metrics from cortex service.
+
+```
+docker run -d --name=grafana -p 3000:3000 grafana/grafana
+```
+
+Add a data source in grafana by picking the prometheus type as data source and configure cortex url to query metrics. 
+i.e http://localhost:9009/api/prom
+
+Now you can monitor the cortex reads & writes by creating the dashboard for cortex.
+
+clone grafonnet project using the below cmd
+
+```
+git clone https://github.com/grafana/grafonnet-lib.git
+```
+
+Now configure the grafonnet path in the below commands. This will generate cortex read, write path dashboard for grafana.
+
+You can find cortex-read.jsonnet at
```

Comment:
Instead of copy-pasting URLs, please always link them. For example:

```
You may use the following Cortex dashboards:

- [Cortex - read](https://github.com/cortexproject/cortex/blob/master/production/dashboards/cortex-read.jsonnet): `jsonnet -J <grafonnet-path> -o cortex-read.json cortex-read.jsonnet`
- [Cortex - write](https://github.com/cortexproject/cortex/blob/master/production/dashboards/cortex-write.jsonnet): `jsonnet -J <grafonnet-path> -o cortex-write.json cortex-write.jsonnet`
```